api/app/helpers/transcript.ts

```diff
@@ -21,4 +23,27 @@ function generateUniqueHash(content: any) {
   return buffer.toString("base64");
 }
 
-export { generateUniqueStr, generateUniqueHash };
+function parseMdToJSON<T extends BaseParsedMdContent>(mdContent: string): T {
+  const regex = /^---\n([\s\S]*?)\n---\n([\s\S]*)$/;
+  const match = mdContent.match(regex);
+
+  if (!match) {
+    throw new Error("Invalid Markdown format");
+  }
+
+  const [, header, body] = match;
+
+  let json: Record<string, string | string[] | undefined>;
+  try {
+    // Use js-yaml's safeLoad function to parse the YAML header
+    json = yaml.load(header) as Partial<T>;
```

Comment:
Why are we type-casting as `Partial<T>`? Is the result sometimes incomplete?

Reply:
Yes, which I think is fine since we are passing an MD file with an undefined structure.